gematria/datasets/basic_block_utils_test.cc

```diff
@@ -112,5 +118,38 @@ TEST_F(BasicBlockUtilsTest, UsedRegistersImplicitUse) {
   EXPECT_THAT(UsedRegisters, UnorderedElementsAre(X86::RAX, X86::RSP));
 }
 
+TEST_F(BasicBlockUtilsTest, LoopRegisterSingleInstruction) {
+  std::optional<unsigned> LoopRegister = getLoopRegister(R"asm(
+    mov %rax, %rcx
+  )asm");
+  EXPECT_EQ(*LoopRegister, X86::RDX);
```

Comment:
This might break when LLVM changes register indices. That doesn't happen very often, but we've already been broken by this a couple of times.

A more reliable way would be to test that the returned value is one of the remaining GPRs.

Reply:
Interesting. Didn't realize LLVM would change the register indices that often. Good point though that this test would be brittle regardless. I've changed it as you've suggested.